cylc/rose/rose.py

```diff
@@ -255,3 +259,125 @@ def rose_config_tree_loader(dir_=None, opts=None):
     )
 
     return config_tree
+
+
+def record_cylc_install_options(
+    dest_path,
+    opt_conf_keys=[],
+    defines=[],
+    suite_defines=[],
+):
+    """Create a file recording Cylc install config options.
+
+    Currently developing on the assumption that I'll be adding a the ability
+    to use a [template variables] section in the config, so that logic
+    identifying the templating section as jinja2 or empy is not req'd.
+
+    Args:
+        dest_path (pathlib.Path | or str)
+        opt_conf_keys (list of str):
+            Equivelent of ``rose suite-run --option KEY``
+        defines (list of str):
+            Equivelent of ``rose suite-run --define KEY=VAL``
+        suite_defines (list of str):
+            Equivelent of ``rose suite-run --define-suite KEY=VAL``
+
+    TODO:
+        - Consider what happens to opts
+
+    """
+    # Construct new ouput based on optional Configs
+    newconfig = []
+    newconfig = ConfigNode()
+    # Currently unclear how Rose deas with these.
+    # Coming back late because treatent of args more obvious.
+    if opt_conf_keys:
+        newconfig.set(
+            ["opts"], ' '.join(opt_conf_keys), ConfigNode.STATE_USER_IGNORED
+        )
```

Comment:
Quick summary of the proposal with relation to opt configs passed in on the CLI:

**1. In the event of duplicates, remove the earliest occurrences.**

Here's an example algorithm that does the reverse (removes the latest occurrences):

```python
def simplify_path(path):
    """Removes duplication in paths whilst maintaining integrity.

    If duplicate items are present in a path this keeps the first item and
    removes any subsequent duplicates.

    Examples:
        >>> simplify_path('')
        ''
        >>> simplify_path('a')
        'a'
        >>> simplify_path('a:a:a')
        'a'
        >>> simplify_path('a:b:a')
        'a:b'
        >>> simplify_path('a:b:b:a')
        'a:b'
        >>> simplify_path('a:b:a:b:c:d:a:b:c:d:e')
        'a:b:c:d:e'

    """
    path = path.split(':')
    counter = Counter(path)
    for item, count in counter.items():
        ptr = len(path) - 1
        while count > 1:
            if path[ptr] == item:
                path.pop(ptr)
                count -= 1
            ptr -= 1
    return ':'.join(path)
```

**2. Add these confs to the `cylc-install` opt config as an ignored variable.**

**3. Append these opts to the end of the list in `rose-suite.conf|opts`.**

Reply:
Thanks.